library/core/class.form.php

```diff
@@ -381,10 +379,17 @@ public function color($fieldName, $options = []) {
 
         // Default starting color for color input. Color inputs require one, Chrome will throw a warning if one
         // doesn't exist. The javascript will override this.
-        $colorAttributes['value'] = '#ffffff';
+        $colorAttributes['value'] = $options['Value'] ?? '#ffffff';
+        $valueAttributes['value'] = $colorAttributes['value'];
```

Comment:
I believe this is gonna overwrite the form data value. For example, if you implemented this for the "Mobile Address Bar Color" field of the "Branding" page in the dashboard, it'll only ever show the starting/default color. The value from the `Garden.MobileAddressBarColor` config will be overwritten when the form is rendered.

Reply:
Astonishing, you are right. But take a look at the markup generated from the current Form->color

~~~
<div id="Garden-dot-MobileAddressBarColor" class="js-color-picker color-picker input-group" data-allow-empty="true">
<input ... id="Form_Garden-dot-MobileAddressBarColor" ... value="#00ff00" ...>
<input ... id="Form_Garden-dot-MobileAddressBarColor-text" value="" ...>
<span class="js-color-picker-preview color-picker-preview" style="background-color: rgb(0, 255, 0);"></span>
<input ... id="Form_Garden-dot-MobileAddressBarColor-color" ... value="#ffffff"...>
</div>
~~~

I've shortened it a bit. The html markup shows that default value of #fff from the Form class in the input#color, whereas the custom color #0f0 is in the first input and the span.
I wouldn't expect to see the default color in the form element and although it doesn't have consequences I think it should be avoided to have different information in one and the same form element.

As far as I can see the process right now is this:
1. the color element is created with the default value #fff only in the input#color
2. the configuration module sets the value from the config to the "main" form input#Form_Garden-dot-MobileAddressBarColor which has the class .js-color-picker-value
3. colorpicker.js picks the value from input.js-color-picker-value and sets it as the text and for the preview color

It definetly is a problem that the js overrides the default color value. But if I see it right that color value is a dummy value, the code shown above proves this. I'll try another approach...
I'll spend some more time on it to understand what is going on there...